### PR TITLE
Fix PSSA about SecureString (Password field)

### PR DESCRIPTION
**Pull request by alagoutte**
Hi,

Fix warning when use PSSA (PowerShell Script Analyzer) on PowerVRNI(.psm1) about password field is unsecure, need to use SecureString type

### CODE REVIEW

File: PowervRNI.psm1 [hunk 1]
```diff
@@ -369,22 +369,26 @@ function Connect-vRNIServer {
   if successful.
 
   .EXAMPLE
-  PS C:\> Connect-vRNIServer -Server vrni-platform.lab.local -Username admin@local -Password secret
+  PS C:\> $mysecpassword = ConvertTo-SecureString secret -AsPlainText -Force
+  PS C:\> Connect-vRNIServer -Server vrni-platform.lab.local -Username admin@local -Password $mysecpassword
   Connect to vRNI Platform VM with the hostname vrni-platform.lab.local
```
Comment (alagoutte) [inline PowervRNI.psm1:374]: it will be better to add some blank line after command ?

File: PowervRNI.psm1 [hunk 2]
```diff
@@ -4077,7 +4090,8 @@ function Set-vRNIUserPassword {
   allow member to set their own passwords.
 
   .EXAMPLE
-  PS C:\> Set-vRNIUserPassword -Username admin@local -NewPassword 'mynewpassword'
+  PS C:\> $mynewpassword = ConvertTo-SecureString mynewpassword -AsPlainText -Force
+  PS C:\> Set-vRNIUserPassword -Username admin@local -NewPassword $mynewpassword
```
Comment (alagoutte) [inline PowervRNI.psm1:4095]: missing description of example...

File: PowervRNI.psm1 [hunk 3]
```diff
@@ -4077,7 +4090,8 @@ function Set-vRNIUserPassword {
...
 
   .EXAMPLE
   PS C:\> Set-vRNIUserPassword -Username admin@local
```
Comment (alagoutte) [inline PowervRNI.psm1:4097]: Why don't use more PS style with pipeline ? ( Get-vRNIUser -UserName admin@local | Set-vRNIUserPassword )

**Comment by smitmartijn** [comment]
This is not backward compatible, please use a new parameter for this (-SecurePassword)

**Comment by alagoutte** [comment]
Hi Martirjn

Sorry, it is mandatory to break the comparability for avoid error (and secure password)

**Comment by smitmartijn** [comment]
No, we can deprecate and remove it after a while, but not instantly - this will break things.

**Comment by alagoutte** [comment]
But, PSSA will be always no happy about unsecure password 
And only add note a release about this break (and i think it is very limited...)

**Comment by smitmartijn** [comment]
That's why I mentioned we can deprecate the old method, but there has to be an overlap. All scheduled tasks will fail if the people do not read the release notes. Add a warning output about the deprecating when -Password is used and make this -SecurePassword.

**Comment by alagoutte** [comment]
ok, i will update my patch on this way...

(but it will fail later when will be remove parameter ;))

**Comment by alagoutte** [comment]
Hi @smitmartijn can you look last change (for Connect-VRNIServer), it is good for you ?

**Comment by smitmartijn** [comment]
Sorry for the delay. I was about to merge it but noticed the same backward compatibility issue is there with New-vRNIDataSource, Update-vRNIDataSource, and Set-vRNIDataSourceSNMPConfig. Could you do the same for those? 

I've also removed Connect-NIBetaServer, so you can take that one out as well.

**Comment by alagoutte** [comment]
> Sorry for the delay. I was about to merge it but noticed the same backward compatibility issue is there with New-vRNIDataSource, Update-vRNIDataSource, and Set-vRNIDataSourceSNMPConfig. Could you do the same for those?
> 
> I've also removed Connect-NIBetaServer, so you can take that one out as well.

Change for Connect-VRNIServer is good for you ? i can make the same for VRNIDataSource ?

**Comment by smitmartijn** [comment]
> Change for Connect-VRNIServer is good for you ? i can make the same for VRNIDataSource ?

Yessir, that looks good!

**Pull request merged**